File: backend/src/main/kotlin/denyyyys/salons_nearby/models/Salon.kt
package denyyyys.salons_nearby.models

import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document
import org.springframework.data.mongodb.core.mapping.Field

@Document(collection = "salons")
data class Salon(
    @Id
    val id: String? = null,
    @Field("booksy_business_id")
    val booksyBusinessId: Long,
    val address: String,
    val description: String,
    val district: String,
    val email: String?,
    @Field("facebook_link")
    val facebookLink: String?,
    @Field("instagram_link")
    val instagramLink: String?,
    val name: String,
    val phone: String?,
    val rating: Double,
    @Field("reviews_count")
    val reviewsCount: Int,
    val services: List<Service> = emptyList()
)

data class Service(
    val name: String? = null,
    @Field("min_price")
    val minPrice: Int = 0,
    @Field("max_price")
    val maxPrice: Int = 0
)

File: frontend/types/salon.ts
export type SalonListItem = {
    booksyBusinessId: number;
    name: string;
    address: string;
    district: string;
    rating: number;
    reviewsCount: number;
    minPrice: number | null;
    maxPrice: number | null;
};

export type GetSalonsParams = {
    page?: number;
    name?: string;
    district?: string;
    serviceType?: string;
};

export type Service = {
    name: string | null;
    minPrice: number;
    maxPrice: number;
};

export type SalonDetails = {
    booksyBusinessId: number;
    address: string;
    description: string;
    district: string;
    email: string | null;
    facebookLink: string | null;
    instagramLink: string | null;
    name: string;
    phone: string | null;
    rating: number;
    reviewsCount: number;
    minPrice: number | null;
    maxPrice: number | null;
    services: Service[];
};
